src/mongocxx/result/replace_one.cpp

```diff
@@ -36,7 +36,11 @@ std::int32_t replace_one::modified_count() const {
 }
 
 stdx::optional<bsoncxx::document::element> replace_one::upserted_id() const {
-    return _result.upserted_ids();
+	if (_result.upserted_ids().size() == 0) {
+		return stdx::nullopt;
+	} else {
```

Comment:
Since you have a return, you don't need the else:

if (...) {
    return ...;
}

return ...